routing/http/client/client.go

```diff
@@ -67,53 +60,72 @@ var defaultUserAgent = moduleVersion()
 
 var _ contentrouter.Client = &Client{}
 
+func newDefaultHTTPClient(userAgent string) *http.Client {
+	return &http.Client{
+		Transport: &ResponseBodyLimitedTransport{
+			RoundTripper: http.DefaultTransport,
+			LimitBytes:   1 << 20,
+			UserAgent:    userAgent,
+		},
+	}
+}
+
 type httpClient interface {
 	Do(req *http.Request) (*http.Response, error)
 }
 
-type Option func(*Client)
+type Option func(*Client) error
 
 func WithIdentity(identity crypto.PrivKey) Option {
-	return func(c *Client) {
+	return func(c *Client) error {
 		c.identity = identity
+		return nil
 	}
 }
 
+// WithHTTPClient sets a custom HTTP Client to be used with [Client].
 func WithHTTPClient(h httpClient) Option {
-	return func(c *Client) {
+	return func(c *Client) error {
 		c.httpClient = h
+		return nil
 	}
 }
 
+// WithUserAgent sets a custom user agent to use with the HTTP Client. This only
```

Comment:
WDYT about specifically noting that this modifies the underlying http.Client (i.e. so that people don't try and pass the same HTTP Client to two routing clients with different user agents and get confused as to why both user agents end up as the second one)